### DISCUSSION TITLE
NextJS + Prisma，why is a 2M WASM file generated?

### DISCUSSION THREAD

**Discussion by tangpanqing** [prisma/prisma#24087] [Q&A]
### Question

Guys, I need your help

I use NextJS + Prisma + CloudFlare D1 to build the app

When uploading and packaging, it actually produced a 2M wasm file, causing the entire package file to be too large and the deployment to be unsuccessful.

First of all, I don't understand why it generates this file? And how to solve this problem?

Thanks

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by janpio** [comment]
The Wasm file is the Prisma Query Engine, which we use under the hood to generate the SQL queries and so on. It is unfortunately a bit big, but usually compresses down well enough that your app still fits into the 1MB limit of Workers on the free account of Cloudflare.

I assume that is not the case for you and you are hitting the upload limit?

The only _real_ solution is to use a paid account to deploy your Cloudflare Workers, as the limit is higher than 1M there. Sorry.

**Comment by kunal00000** [comment]
Experiencing the same issue with the following versions:
`prisma`: `^5.16.1`
`@cloudflare/next-on-pages`: `^1.12.0`

Database: Cloudflare D1 (sqlite)

<img width="898" alt="Screenshot 2024-07-03 at 2 13 49 AM" src="https://github.com/prisma/prisma/assets/92316166/d7521c25-6238-41ff-a32d-00650af0b0b1">

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
